internal/pubsub/pub_test.go

```diff
@@ -0,0 +1,62 @@
+package pubsub_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/google/go-cloud/internal/pubsub"
+	"github.com/google/go-cloud/internal/pubsub/driver"
+)
+
+type funcTopic struct {
+	sendBatch func(ctx context.Context, ms []*driver.Message) error
+}
+
+func (t *funcTopic) SendBatch(ctx context.Context, ms []*driver.Message) error {
+	return t.sendBatch(ctx, ms)
+}
+
+func (t *funcTopic) Close() error {
+	return nil
+}
+
+func (s *funcTopic) IsRetryable(error) bool { return false }
+
+func (s *funcTopic) As(i interface{}) bool { return false }
+
+func TestTopicShutdownCanBeCanceledEvenWithHangingSend(t *testing.T) {
```

Comment:
Consider adding this test in a separate PR, it is not related to the rename.

Reply:
The rename isn't just a rename. The new name reflects new functionality, and this test demonstrates the new functionality.